src/base/http/requestparser.cpp

```diff
@@ -30,314 +31,322 @@
 #include "requestparser.h"
 
 #include <QDebug>
-#include <QDir>
+#include <QRegularExpression>
 #include <QStringList>
 #include <QUrl>
 #include <QUrlQuery>
 
-const QByteArray EOL("\r\n");
-const QByteArray EOH("\r\n\r\n");
+#include "base/utils/string.h"
 
-inline QString unquoted(const QString &str)
+using namespace Http;
+using QStringPair = QPair<QString, QString>;
+
+namespace
 {
-    if ((str[0] == '\"') && (str[str.length() - 1] == '\"'))
-        return str.mid(1, str.length() - 2);
+    const QByteArray EOH = QByteArray(CRLF).repeated(2);
 
-    return str;
-}
+    QByteArray chopIfEndsWith(QByteArray in, const QByteArray &str)
+    {
+        if (in.endsWith(str))
+            in.chop(str.size());
+        return in;
+    }
 
-using namespace Http;
+    QList<QByteArray> split(const QByteArray &in, const QByteArray &sep, const QString::SplitBehavior behavior = QString::KeepEmptyParts)
+    {
+        // mimic QString::split(sep, behavior)
 
-RequestParser::ErrorCode RequestParser::parse(const QByteArray &data, Request &request, uint maxContentLength)
-{
-    return RequestParser(maxContentLength).parseHttpRequest(data, request);
+        if (sep.isEmpty())
+            return {in};
+
+        QList<QByteArray> ret;
+
+        int head = 0;
+        while (head < in.size()) {
+            int end = in.indexOf(sep, head);
+            if (end < 0)
+                end = in.size();
+
+            // omit empty parts
+            const QByteArray part((in.constData() + head), (end - head));
+            if (!part.isEmpty() || (behavior == QString::KeepEmptyParts))
+                ret += part;
+
+            head = end + sep.size();
+        }
+
+        return ret;
+    }
+
+    QByteArray viewMid(const QByteArray &in, const int pos, const int len = -1)
```

Comment:
Shouldn't it return const QByteArray? Otherwise it allows to modify const object (in QByteArray is const).

Reply:
yes, fixed.